opencti-platform/opencti-front/src/private/components/settings/sub_types/EntitySettingScale.tsx

```diff
@@ -0,0 +1,221 @@
+import makeStyles from '@mui/styles/makeStyles';
+import React, { useState, FunctionComponent } from 'react';
+import * as Yup from 'yup';
+import { Add } from '@mui/icons-material';
+import IconButton from '@mui/material/IconButton';
+import Typography from '@mui/material/Typography';
+import { Button } from '@mui/material';
+import { useFormatter } from '../../../../components/i18n';
+import { MESSAGING$ } from '../../../../relay/environment';
+import type { ScaleConfig, Tick, UndefinedTick } from './EntitySettingAttributesConfigurationScale';
+import EntitySettingScaleTickLine from './EntitySettingScaleTickLine';
+
+const useStyles = makeStyles(() => ({
+  container: {
+    padding: '10px 20px 20px 20px',
+  },
+}));
+
+interface EntitySettingScaleProps {
+  scaleConfig: ScaleConfig,
+  submitting: boolean,
+  handleSubmit: (scale: ScaleConfig) => void
+}
+
+const isTickDefinitionValid = (tickDefinition: ScaleConfig) => {
+  let isValid = true;
+  if (tickDefinition.min.value > tickDefinition.max.value) {
+    MESSAGING$.notifyError('The min value cannot be greater than max value');
+    return false;
+  }
+  tickDefinition.ticks.forEach((tick) => {
+    if (!!tick.value && (tick.value < tickDefinition.min.value || tick.value > tickDefinition.max.value)) {
+      MESSAGING$.notifyError('Each tick value must be between min and max value');
+      isValid = false;
+    }
+  });
+  return isValid;
+};
+
+const EntitySettingScale: FunctionComponent<EntitySettingScaleProps> = ({
+  scaleConfig,
+  submitting,
+  handleSubmit,
+}) => {
+  const { t } = useFormatter();
+  const classes = useStyles();
+  const [tickDefinition, setTickDefinition] = useState<ScaleConfig>(scaleConfig);
+
+  const { min, max } = tickDefinition;
+
+  const tickValidation = () => {
+    return Yup.object()
+      .shape({
+        label: Yup.string().required(t('This field is required')),
+        color: Yup.string().required(t('This field is required')),
+        value: Yup.number()
+          .required(t('This field is required'))
+          .test('SchemaValidation', t('The value must be between min and max value'), (value) => {
+            return !(!!value && (value < tickDefinition.min.value || value > tickDefinition.max.value));
+          }),
+      });
+  };
+
+  const minMaxValidation = () => {
+    return Yup.object()
+      .shape({
+        label: Yup.string().required(t('This field is required')),
+        color: Yup.string().required(t('This field is required')),
```

Comment:
We can add the required pattern defined in back-end

Reply:
Yes, but if we add it here, the Yup validation can throw an exception here and we don't have to call the back-end to have it.
This removes an extra call to the server.

Reply:
Done